src/decorators.ts

```diff
@@ -44,7 +52,20 @@ export function property(options?: PropertyOptions) {
     const reflectToAttribute: boolean =
         options && options.reflectToAttribute || false;
     const readOnly: boolean = options && options.readOnly || false;
-    const type = Reflect.getMetadata('design:type', proto, propName);
+
+    let type;
+    if (options && options.hasOwnProperty('type')) {
+      type = options.type;
+    } else if (
+        (window as any).Reflect && Reflect.hasMetadata && Reflect.getMetadata &&
+        Reflect.hasMetadata('design:type', proto, propName)) {
+      type = Reflect.getMetadata('design:type', proto, propName);
+    } else {
+      console.error(
+          'A type could not be found for ${propName}. ' +
+          'Set a type or configure Metadata Reflection API support.');
```

Comment:
Still kind of on the fence here, because you might just not care about de-serialization. Fine for now, but let's get feedback from TS users.